### PR TITLE
Added distance unit selector

### PR DESCRIPTION
**Pull request by Ninjasoturi**
In search box the distance to a point is displayed in miles even though the unit says km. Instead of fixing that to either of the units, I added `distanceUnit` in config so admins can select which ever they prefer. I guess the most optimal would be to have it possible for users to select their preferred unit but making that happen was way beyond my skill level.

### CODE REVIEW

**Comment by TurtIeSocks** [comment]
This is awesome, thank you! Going to test tonight.

File: server/src/services/DbCheck.js [hunk 1]
```diff
@@ -36,10 +36,11 @@ module.exports = class DbCheck {
           },
         })
       })
+    this.distanceUnit = distanceUnit
   }
 
-  static getDistance(args, isMad) {
-    return raw(`ROUND(( 3959 * acos( cos( radians(${args.lat}) ) * cos( radians( ${isMad ? 'latitude' : 'lat'} ) ) * cos( radians( ${isMad ? 'longitude' : 'lon'} ) - radians(${args.lon}) ) + sin( radians(${args.lat}) ) * sin( radians( ${isMad ? 'latitude' : 'lat'} ) ) ) ),2)`).as('distance')
+  static getDistance(args, isMad, distanceUnit) {
+    return raw(`ROUND(( ${(distanceUnit == 'mi') ? '3959' : '6371'} * acos( cos( radians(${args.lat}) ) * cos( radians( ${isMad ? 'latitude' : 'lat'} ) ) * cos( radians( ${isMad ? 'longitude' : 'lon'} ) - radians(${args.lon}) ) + sin( radians(${args.lat}) ) * sin( radians( ${isMad ? 'latitude' : 'lat'} ) ) ) ),2)`).as('distance')
```
Comment (TurtIeSocks) [inline server/src/services/DbCheck.js:43]:
```suggestion
    return raw(`ROUND(( ${distanceUnit === 'mi' ? '3959' : '6371'} * acos( cos( radians(${args.lat}) ) * cos( radians( ${isMad ? 'latitude' : 'lat'} ) ) * cos( radians( ${isMad ? 'longitude' : 'lon'} ) - radians(${args.lon}) ) + sin( radians(${args.lat}) ) * sin( radians( ${isMad ? 'latitude' : 'lat'} ) ) ) ),2)`).as('distance')
```

File: src/components/layout/dialogs/Search.jsx [hunk 2]
```diff
@@ -192,7 +192,7 @@ export default function Search({
               )}
             </Grid>
             <Grid item xs={2} style={{ textAlign: 'center' }}>
-              <Typography variant="caption">{option.distance}{t('km')}</Typography>
+              <Typography variant="caption">{option.distance} {(map.distanceUnit == 'mi' ? t('mi') : t('km'))}</Typography>
```
Comment (TurtIeSocks) [inline src/components/layout/dialogs/Search.jsx:195]:
```suggestion
              <Typography variant="caption">{option.distance} {map.distanceUnit === 'mi' ? t('mi') : t('km')}</Typography>
```

**Review by TurtIeSocks — CHANGES_REQUESTED** [review]
Just a couple of stylistic code changes but otherwise looks and works great, thank you!

File: server/src/services/DbCheck.js (hunk 1 above)
Comment (Ninjasoturi) [reply server/src/services/DbCheck.js]: Done. I also applied same changes to the changes in `src/components/layout/dialogs/Search.jsx`

**Review by TurtIeSocks — APPROVED** [review]

**Comment by Mygod** [comment]
What the fucking shit is this? This is why old people need to die.

**Comment by Mygod** [comment]
This option needs to die. Literally everything in the game uses (kilo)meters, notably routes which is user visible. RM also doesn't respect this option for routes.

**Comment by roundaboutluke** [comment]
> What the fucking shit is this? This is why old people need to die.

England has entered the chat

**Pull request merged**